src/url.cpp

```diff
@@ -469,6 +474,13 @@ namespace ada {
     if(checkers::is_ipv4(host.value())) {
       return parse_ipv4(host.value());
     }
+
+    // The domain label size ehckers should only be applied for fully qualified domain names.
+    // FQDNs have label hierarchy (meaning at least one dot).
+    if(host.value().find_first_of(".") != std::string_view::npos) {
```

Comment:
Possibly `host.value().find_first_of(".")` to `host.value().find_first_of('.')`. I don't know that it actually makes a difference... but I expect that '.' might be faster (???). I also tend to prefer "find('.')" because it is shorter... though it might not matter.

Reply:
(This is not a request for change... your `host.value().find_first_of(".")` is probably? fine.)